src/coord/src/coord.rs

```diff
@@ -242,6 +242,7 @@ pub struct Config {
     pub metrics_registry: MetricsRegistry,
     pub now: NowFn,
     pub secrets_controller: Box<dyn SecretsController>,
+    pub availability_zones: Vec<String>,
```

Comment:
#11809 mentions that the AZ from `CREATE REPLICA` should be verified against the set of allowed AZs in the coordinator. It looks like we could also do this checking in the compute controller, as #11884 does for the replica size. Are there any benefits from doing one over the other?

Reply:
I think if you do the validation in the compute controller, it is too late, and failures will cause a crash. We'll need to shake out a saner story for this eventually. Maybe the controller wants to expose an explicit validation step, which we do before we commit the cluster record to the catalog, and then declare that cluster creation with a validated cluster config _cannot_ fail.